evaluate/task/task-transpile_test.go

```diff
@@ -546,3 +476,114 @@ func TestValidateTranspileRepository(t *testing.T) {
 		})
 	})
 }
+
+func TestValidateImplementationDirectory(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Before func(implementationDirectory string)
+
+		ExpectedErrContains func(error error)
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			logOutput, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Logf("Logging output: %s", logOutput.String())
+				}
+			}()
+
+			temporaryDirectory := t.TempDir()
+
+			implementationDirectoryPath := filepath.Join(temporaryDirectory, "implementation")
+			require.NoError(t, osutil.MkdirAll(implementationDirectoryPath))
+
+			if tc.Before != nil {
+				tc.Before(implementationDirectoryPath)
+			}
+
+			actualErr := validateImplementationDirectory(logger, implementationDirectoryPath)
+			if tc.ExpectedErrContains != nil {
+				tc.ExpectedErrContains(actualErr)
+			} else {
+				require.NoError(t, actualErr)
+			}
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Implementation directory contains files",
+
+		Before: func(implementationDirectory string) {
+			require.NoError(t, os.WriteFile(filepath.Join(implementationDirectory, "file.go"), []byte("content"), 0700))
+		},
+
+		ExpectedErrContains: func(err error) {
+			assert.ErrorContains(t, err, "must contain only subdirectories with source code files to transpile, but found one file: \"file.go\"")
+		},
+	})
+	validate(t, &testCase{
+		Name: "Malformed subdirectory name",
+
+		Before: func(implementationDirectory string) {
+			require.NoError(t, osutil.MkdirAll(filepath.Join(implementationDirectory, "malformed-directory-name")))
+		},
+
+		ExpectedErrContains: func(err error) {
+			assert.ErrorContains(t, err, "the implementation subdirectories names must be in the form \"implementation-<language>\", but found \"malformed-directory-name\"")
+		},
+	})
+	validate(t, &testCase{
+		Name: "Unsupported language",
+
+		Before: func(implementationDirectory string) {
+			require.NoError(t, osutil.MkdirAll(filepath.Join(implementationDirectory, "implementation-haskell")))
```

Comment:
Plz call it `implementation-someunsupportedlanguage` cause hopefully there will never be `someunsupportedlanguage` but once we support `haskell` then this test will fail.